Add missing types to property mapping documentation (#7609)
doctrine/doctrine2: docs/en/reference/basic-mapping.rst

depending on the RDBMS brand). Here is a quick overview of the
built-in mapping types:

-  ``string``: Type that maps an SQL VARCHAR to a PHP string.
-  ``integer``: Type that maps an SQL INT to a PHP integer.

> -  ``smallint``: Type that maps an database SMALLINT to a PHP

Reviewer: That was probably from a proponent of the /ˈsiːkwəl/  pronunciation :)

Author: A decision should be made so we can have all types in the documentation.

Reviewer 2: "a database", not "an database"

Reviewer: ```suggestion
-  ``smallint``: Type that maps a database SMALLINT to a PHP
```

   integer.
-  ``bigint``: Type that maps a database BIGINT to a PHP string.
-  ``boolean``: Type that maps an SQL boolean or equivalent (TINYINT) to a PHP boolean.
-  ``decimal``: Type that maps an SQL DECIMAL to a PHP string.
-  ``date``: Type that maps an SQL DATETIME to a PHP DateTime
   object.
-  ``date_immutable``: Type that maps an SQL DATETIME to a PHP DateTimeImmutable
   object.
-  ``time``: Type that maps an SQL TIME to a PHP DateTime object.
-  ``time_immutable``: Type that maps an SQL TIME to a PHP DateTimeImmutable object.
-  ``datetime``: Type that maps an SQL DATETIME/TIMESTAMP to a PHP DateTime
   object with the current timezone.
-  ``datetimetz``: Type that maps an SQL DATETIME/TIMESTAMP to a PHP DateTime
   object with the timezone specified in the value from the database.
-  ``datetime_immutable``: Type that maps an SQL DATETIME/TIMESTAMP to a PHP DateTimeImmutable
   object with the current timezone.

> -  ``datetimetz_immutable``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP DateTimeImmutable

Reviewer: ```suggestion
-  ``datetimetz_immutable``: Type that maps an SQL DATETIME/TIMESTAMP to a PHP DateTimeImmutable
```

   object with the timezone specified in the value from the database.
-  ``dateinterval``: Type that maps an interval to a PHP DateInterval object
-  ``text``: Type that maps an SQL CLOB to a PHP string.
-  ``object``: Type that maps an SQL CLOB to a PHP object using
   ``serialize()`` and ``unserialize()``
-  ``array``: Type that maps an SQL CLOB to a PHP array using
   ``serialize()`` and ``unserialize()``
-  ``simple_array``: Type that maps an SQL CLOB to a one-dimensional PHP array using
   ``implode()`` and ``explode()``, with a comma as delimiter. *IMPORTANT*
   Only use this type if you are sure that your values cannot contain a ",".
-  ``json_array``: Type that maps an SQL CLOB to a PHP array using

>    ``json_encode()`` and ``json_decode()``. This one was deprecated in favor

Reviewer 2: has been deprecated

Reviewer: ```suggestion
   ``json_encode()`` and ``json_decode()``. This one has been deprecated in favor
```

   of ``json`` type.
-  ``json``: Type that maps an SQL CLOB to a PHP array using
   ``json_encode()`` and ``json_decode()``. An empty value is correctly represented as ``null``
-  ``float``: Type that maps an SQL Float (Double Precision) to a
   PHP double. *IMPORTANT*: Works only with locale settings that use
   decimal points as separator.

> -  ``guid``: Type that maps an database GUID/UUID to a PHP string. Defaults to

Reviewer 2: "a database", not "an database"

Reviewer: ```suggestion
-  ``guid``: Type that maps a database GUID/UUID to a PHP string. Defaults to
```

-  ``decimal``: Type that maps a SQL DECIMAL to a PHP string.
-  ``date``: Type that maps a SQL DATETIME to a PHP DateTime
   object.

> -  ``date_immutable``: Type that maps a SQL DATETIME to a PHP DateTimeImmutable

Reviewer 3: In https://github.com/doctrine/dbal/pull/2793, it was agreed that it's `[ˈɛs kjuː ˈɛl]` (therefore, "an") not `[ˈsiːkwəl]` (not "a").

   object.
-  ``time``: Type that maps a SQL TIME to a PHP DateTime object.
-  ``time_immutable``: Type that maps a SQL TIME to a PHP DateTimeImmutable object.
-  ``datetime``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
   DateTime object.
-  ``datetime_immutable``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
   DateTimeImmutable object.
-  ``datetimetz``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
   DateTime object with timezone.
-  ``datetimetz_immutable``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP

>    DateTimeImmutable object with timezone.

Reviewer 3: `DateTimeImmutable` always has a timezone. The difference is whether it's taken from the DB value or from the server configuration. Otherwise, the wording looks a bit misleading.

Author: So is the `datetimetz` one. Do you have a suggestion what would be more understandable for a reader? I'll change the description for both.

Reviewer 3: - `datetime_immutable`: Type that maps a SQL DATETIME/TIMESTAMP to a PHP `DateTimeImmutable` object with the current timezone.
- `datetimetz_immutable`: Type that maps a SQL DATETIME/TIMESTAMP to a PHP `DateTimeImmutable` object with the timezone specified in the value from the database.

Author: Thank you.

   ``serialize()`` and ``unserialize()``
-  ``array``: Type that maps a SQL CLOB to a PHP array using
   ``serialize()`` and ``unserialize()``
-  ``simple_array``: Type that maps a SQL CLOB to a one-dimensional PHP array using
   ``implode()`` and ``explode()``, with a comma as delimiter. *IMPORTANT*
   Only use this type if you are sure that your values cannot contain a ",".
-  ``json_array``: Type that maps a SQL CLOB to a PHP array using
   ``json_encode()`` and ``json_decode()``. This one was deprecated in favor
   of ``json`` type.
-  ``json``: Type that maps a SQL CLOB to a PHP array using
   ``json_encode()`` and ``json_decode()``. An empty value is correctly represented as ``null``
-  ``float``: Type that maps a SQL Float (Double Precision) to a
   PHP double. *IMPORTANT*: Works only with locale settings that use
   decimal points as separator.
-  ``guid``: Type that maps a database GUID/UUID to a PHP string. Defaults to
   varchar but uses a specific type if the platform supports it.
-  ``blob``: Type that maps a SQL BLOB to a PHP resource stream
-  ``binary``: Type that maps a SQL binary to a PHP resource stream

> -  ``dateinterval``: Type that maps an interval to a PHP DateInterval object

Reviewer 3: Should this type reside somewhere closer to the rest of the date-related ones?

Author: Makes sense to me.
